### ISSUE TITLE
Exception with Numpy 1.24.0 onwards due to Expired Deprecation of np.int

### ISSUE THREAD

**Issue by thomasWeise** [pdfo/pdfo#55]
Hi.

I am a user of your great library and, first of all, want to thank you for making it available. It is really great.

Recently, I am getting exceptions and I have tracked them down to the expired deprecation of [`np.int`](https://github.com/numpy/numpy/pull/22607) in Numpy. This happens in release [`1.24.0`](https://numpy.org/doc/1.24/release/1.24.0-notes.html#expired-deprecations) and I am experiencing it with version Numpy [ `1.24.3`](https://pypi.org/project/numpy/1.24.3) and pdfo version 1.3.

The exception reads as:

```
raise AttributeError(__former_attrs__[attr])
AttributeError: module 'numpy' has no attribute 'int'.
`np.int` was a deprecated alias for the builtin `int`. To avoid this error in existing code, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
``` 

It happens in `_dependencies.py` at line 1749, which says:

```
    validated = False   # 1747
    if 'npt' in option_fields:  # 1748
        integer_types = (int, np.int, np.int8, np.int16, np.int32, np.int64)  # <--- HERE; 1749
``` 

I think this could probably be fixed rather easily in a backwards compatible way, maybe with a

```
integer_types = [int, np.int8, np.int16, np.int32, np.int64]
if hasattr(np, "int"):
    integer_types.append(np.int)
```

I am not sure whether this is the only issue with the new numpy version, though...

All the best,
   Thomas.

**Comment by ragonneau** [comment]
Hi @thomasWeise,

First, thank you very much for your detailed issue.

I patched this problem in the development version of PDFO (see 508369b372367d4549337c9cba62a5e46ae0e53d). I will make more tests to check whether other deprecated stuff from NumPy is used elsewhere. I will upload the patch on PyPI afterward.

Regards,
Tom.

**Comment by thomasWeise** [comment]
Cool!!

Thank you for this very quick response!

Cheers,
   Thomas.

**Comment by ragonneau** [comment]
Hi Thomas,

I just published version 1.3.1 on PyPI. I checked for other deprecation and did not find any. Please feel free to reopen this issue if I missed any.

Cheers,
Tom.
